x-pack/platform/plugins/private/data_visualizer/public/lite/file_upload_lite_view.tsx

```diff
@@ -71,10 +73,11 @@ export const FileUploadLiteView: FC<Props> = ({
         http,
         dataStart.dataViews,
         autoAddInference ?? null,
+        autoCreateDataView,
         true,
         indexSettings
       ),
-    [autoAddInference, dataStart.dataViews, fileUpload, http, indexSettings]
+    [autoAddInference, autoCreateDataView, dataStart.dataViews, fileUpload, http, indexSettings]
```

Comment:
❓Should we recreate the `FileManager` instance when `autoCreateDataView` changes? Would a setter be better than passing it to the constructor?

Reply:
These settings shouldn't change. The flyout is opened with these initial settings and that defines the mode the upload flyout is in.
In the future we may add a checkbox for this setting, in which case, yes we'd need a setter to allow it to change.